### PR TITLE
DEV-8048 Status of Funds scaffolding

### PR DESCRIPTION
**Pull request by ebdabbs**
**High level description:**

Basic setup for the Status of Funds section to prevent conflicts.

**JIRA Ticket:**
[DEV-8048](https://federal-spending-transparency.atlassian.net/browse/DEV-8048)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [ ] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [ ] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated

Reviewer(s):
- [ ] Code review complete
- [x] #2771 merged
- [x] Base branch updated to `qat`

### CODE REVIEW

File: src/js/components/agencyV2/statusOfFunds/StatusOfFunds.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                            ) : ''}
+                            {level > 0 ? (
+                                <button onClick={() => setLevel(level - 1)}>
+                                    Up
+                                </button>
+                            ) : ''}
+                        </div>
+                    </FlexGridCol>
+                    <FlexGridCol className="status-of-funds__visualization" tablet={9}>
+                        DEV-8049 horizontal bar chart viewing {levels[level]}s
```
Comment (brett-varney) [inline src/js/components/agencyV2/statusOfFunds/StatusOfFunds.jsx:44]: And this the VisualizationSection ?

**Review by brett-varney — CHANGES_REQUESTED** [review]
Casual look: did you abstract components but accidentally left the original code?

**Comment by brett-varney** [comment]
I'm getting errors in WinIE only, seems to start with:

Error fetching active periods:  Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
   "Error fetching active periods: "
   {
      [functions]: ,
      __proto__: { },
      description: "Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.",
      message: "Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.",
      name: "Error",
      stack: "Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
   at checkForNestedUpdates (eval code:23093:9)
   at scheduleUpdateOnFiber (eval code:21164:3)
   at classComponentUpdater.enqueueSetState (eval code:12639:5)
   at Component.prototype.setState (eval code:471:3)
   at value (eval code:11:73251)
   at invokeFunc (eval code:10401:9)
   at debounced (eval code:10484:13)
   at value (eval code:11:71715)
   at commitLifeCycles (eval code:19835:13)
   at commitLayoutEffects (eval code:22803:7)",
      Symbol()_6.3zvzhqjslwb: undefined,
      Symbol(__immutablehash__)_15.3zvzhqjslp: undefined,
      Symbol(react.async_mode)_14.3zvzhqjslp: undefined,
      Symbol(react.block)_10.3zvzhqjslp: undefined,
      Symbol(react.concurrent_mode)_u.3zvzhqjslpu: undefined,
      Symbol(react.context)_t.3zvzhqjslpu: undefined,
      Symbol(react.element)_n.3zvzhqjslpu: undefined,
      Symbol(react.forward_ref)_v.3zvzhqjslpu: undefined,
      Symbol(react.fragment)_p.3zvzhqjslpu: undefined,
      Symbol(react.fundamental)_11.3zvzhqjslp: undefined,
      Symbol(react.lazy)_z.3zvzhqjslpu: undefined,
      Symbol(react.memo)_y.3zvzhqjslpu: undefined,
      Symbol(react.portal)_o.3zvzhqjslpu: undefined,
      Symbol(react.profiler)_r.3zvzhqjslpu: undefined,
      Symbol(react.provider)_s.3zvzhqjslpu: undefined,
      Symbol(react.responder)_12.3zvzhqjslp: undefined,
      Symbol(react.scope)_13.3zvzhqjslp: undefined,
      Symbol(react.strict_mode)_q.3zvzhqjslpu: undefined,
      Symbol(react.suspense)_w.3zvzhqjslpu: undefined,
      Symbol(react.suspense_list)_x.3zvzhqjslpu: undefined
   }

**Comment by brett-varney** [comment]
Sidebar layout isn't working for me (Win 10), all browsers tested:

image

File: src/js/components/agencyV2/statusOfFunds/StatusOfFunds.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                                <button onClick={() => setLevel(level + 1)}>
+                                    Down
+                                </button>
+                            ) : ''}
+                            {level > 0 ? (
+                                <button onClick={() => setLevel(level - 1)}>
+                                    Up
+                                </button>
+                            ) : ''}
+                        </div>
```
[the remark this answers is not in the document]
Comment (ebdabbs) [reply src/js/components/agencyV2/statusOfFunds/StatusOfFunds.jsx:41]: Yes, thank you!

**Comment by brett-varney** [comment]
@ebdabbs I'm still seeing the two main issues noted (sidebar layout & IE page vanishing with errors)

**Comment by brett-varney** [comment]
Sorry, NOW I see the layout is fine.

**Review by brett-varney — APPROVED** [review]
disclaimer: IE is throwing errors for me (win10), but so does qat branch now, so this doesn't create new issues

**Pull request merged**